### PR TITLE
chore: consolidate all ABI Sway programs

### PR DESCRIPTION
**Pull request by petertonysmith94**
> Part of TS-599, TS-600

- Part of #3031
- Part of #3032

# Summary

- Consolidates all [`abi-typegen/*`](https://github.com/FuelLabs/fuels-ts/blob/master/packages/abi-typegen/test/fixtures/forc-projects) and [`abi-coder/exhaustive-example`](https://github.com/FuelLabs/fuels-ts/blob/master/packages/abi-coder/test/fixtures/forc-projects/exhaustive-examples/src/main.sw) Sway programs into `abi-contract`, `abi-predicate` and `abi-script`.
- Added two lists below which are the original test location (with method) and their destination location.

  <details>
    <summary>Abi Typegen</summary>
    
  
  abi-typegen/tests | abi-contract | Notes
  -- | -- | --
  array-of-enums | types_struct_with_array_of_enums |  
  array-with-generics.simple | types_struct_with_nested_array |  
  array-with-generics.with_generics | types_array_with_generic_struct |  
  bytes | types_bytes |  
  enum-of-enums | types_enum_with_structs | Has a nested enum with this enum
  enum-of-structs | types_enum_with_structs | Has nested struct within enum
  enum-simple | types_enum_with_builtin_type |  
  enum-simple-native | types_enum |  
  evm-address | types_evm_address |  
  fn-void | types_void |  
  full | The backbones of this contract | Copy + pasted this base contract as the initial starting point for abi-contract.
  libs-for-testing | abi-library |  
  minimal | multi_arg_str_str |  
  minimal-with-configurables | configurables |  
  option-simple | types_option |  
  predicate | abi-predicate |  
  predicate-with-configurables | abi-predicate |  
  raw-slice | types_raw_slice |  
  script | abi-script |  
  script-with-configurables | abi-script |  
  std-string | types_std_string |  
  struct-nested.single_params | types_struct_with_multiple_struct_params |  
  struct-nested.multi_params | types_struct_with_complex_nested_struct |  
  struct-simple | types_struct_with_nested_struct |  
  struct-with-array | types_struct_with_implicit_generics |  
  tuple-simple.single_param | types_struct_with_nested_tuple |  
  tuple-simple.tuple_params | types_tuple_complex |  
  vector-simple | types_vector_u8 |  
  </details>

  <details>
    <summary>Exhaustive example</summary>
    
    Exhaustive example | abi-contract |  
    -- | -- | --
    test_function | types_bool |  
    u_8 | types_u8 |  
    u_16 | types_u16 |  
    u_32 | types_u32 |  
    u_64 | types_u64 |  
    string | types_str |  
    boolean | types_bool |  
    b_256 | types_b256 |  
    b_512 | types_b512 |  
    two_args | types_struct_simple |  
    struct_simple | types_struct_generic |  
    struct_generic_simple | types_struct_with_tuple |  
    struct_with_tuple | types_struct_with_implicit_generics |  
    struct_with_implicit_generics | types_bytes |  
    bytes | types_raw_slice |  
    raw_slice | types_std_string |  
    dynamic_string | types_asset_id |  
    asset_id | types_tuple_complex |  
    tuple_as_param | types_array |  
    array_simple | types_array_struct |  
    array_struct | types_struct_with_array |  
    vector_boolean | types_vector_u8 |  
    vector_u8 | types_vector_boolean |  
    arg_then_vector_u8 | types_vector_u8 |  
    vector_u8_then_arg | types_vector_inside_vector |  
    two_u8_vectors | types_vector_inside_array |  
    u32_then_three_vectors_u64 | types_enum_with_vector |  
    enum_simple | types_struct_with_vector |  
    enum_with_builtin_type | types_enum |  
    enum_with_structs | types_enum_with_builtin_type |  
    option_u8 | types_enum_with_structs |  
    return_configurables | types_option_u8 |  
    simple_vector | multi_arg_b256_bool |  
    vector_inside_vector | multi_arg_u32_vector_vector |  
    vector_inside_array | multi_arg_struct_vector |  
    vector_inside_enum | multi_arg_vector_b256 |  
    vector_inside_struct | multi_arg_vector_vector |  
    array_with_generic_struct | types_struct_with_array |  
    entry_one | types_u64 | 🍎 Need to regenerate the fn signatures with new names
    sum | multi_arg_u64_u64 | 🍎 Need to regenerate the fn signatures with new names
    sum_test | 🍎 TODO | 🍎 Need to regenerate the fn signatures with new names
    takes_array | types_array | 🍎 Need to regenerate the fn signatures with new names
    take_enum | types_enum + types_bool | 🍎 Need to regenerate the fn signatures with new names
    my_struct | multi_arg_u64_struct | 🍎 Need to regenerate the fn signatures with new names
    array_of_structs | types_array_struct + types_str | 🍎 Need to regenerate the fn signatures with new names
    complex_function | multi_arg_complex | 🍎 Need to regenerate the fn signatures with new names

  </details>

# Checklist

- [X] All **changes** are **covered** by **tests** (or not applicable)
- [X] All **changes** are **documented** (or not applicable)
- [X] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [X] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

**Comment by petertonysmith94** [comment]
@nedsalk could you give this a pre-review please 🙏🏼 

If you're happy put it in ready for review :)

File: packages/fuel-gauge/test/fixtures/forc-projects/abi-contract/Forc.toml [hunk 1]
```diff
@@ -2,3 +2,6 @@
 authors = ["Fuel Labs <contact@fuel.sh>"]
 license = "Apache-2.0"
 name = "abi-contract"
+
+[dependencies]
+abi-library = { path = "../abi-library" }
```
Comment (petertonysmith94) [inline packages/fuel-gauge/test/fixtures/forc-projects/abi-contract/Forc.toml:7]: Also don't know what is causing the following error with `pnpm forc:check`:

```console
error: Improperly formatted manifest file: /fuel/fuels-ts/packages/fuel-gauge/test/fixtures/forc-projects/abi-contract/Forc.toml
[project]
authors = ["Fuel Labs <contact@fuel.sh>"]
license = "Apache-2.0"
name = "abi-contract"

[dependencies]
abi-library = { path = "../abi-library" }
error: Formatting skipped due to error.
error: Files contain formatting violations.
```

**Comment by nedsalk** [comment]
LGTM, it's exhaustive and possibly repetitive, so we can maybe slim it down as we write tests with it and notice possible repetitions.

File: packages/fuel-gauge/test/fixtures/forc-projects/abi-contract/Forc.toml (hunk 1 above)
Comment (petertonysmith94) [inline packages/fuel-gauge/test/fixtures/forc-projects/abi-contract/Forc.toml:7]:
```suggestion
abi-library = { path = "../abi-library" }

```

Much wow

**Review by danielbate — APPROVED** [review]
Wondeful work, we should definitely refactor other `fuel-gauge` tests to use this, speed up `pretest`

**Pull request merged**